src/core_modules/capture-core/components/D2Form/field/Components/OptionSetField/withOptionsIconElement.js

```diff
@@ -50,11 +51,13 @@ export default () =>
                         <div
                             className={classes.iconContainer}
                         >
-                            <img
-                                style={{ backgroundColor: icon.color }}
-                                className={classes.icon}
-                                src={icon.data}
-                                alt={label}
+                            <NonBundledDhis2Icon
+                                name={icon.name}
+                                color={icon.color}
+                                width={22}
+                                height={22}
+                                cornerRadius={2}
+                                alternativeText={label || undefined}
```

Comment:
Hey, I am wondering if `undefined` is the best value to pass to this prop. How about passing `''`?

Reply:
I believe what I was thinking was that if we pass undefined we could potentially utilize any default values in the inner components. I'm not sure what that default value would be though. What do you think?

Reply:
Great idea! I think that the icon name is a good candidate to be used as the default value for the `alternativeText`. In `NonBundledDhis2Icon.component.js` I have written a deconstructor with a default value for alternativeText. 
WDYT?

Reply:
Perfect! Makes sense. Suggestion applied.